File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.go
type MyStack struct {
	queue []int
}


func Constructor() MyStack {
	return MyStack{
		queue: make([]int, 0),
	}
}


func (this *MyStack) Push(x int)  {
	this.queue = append(this.queue, x)
}


func (this *MyStack) Pop() int {
	val := this.queue[len(this.queue)-1]
    this.queue = this.queue[:len(this.queue)-1]
	return val
}


func (this *MyStack) Top() int {
	return this.queue[len(this.queue)-1]
}


func (this *MyStack) Empty() bool {
	return len(this.queue) == 0
}


/**
 * Your MyStack object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Top();
 * param_4 := obj.Empty();
 */

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.java
 class MyStack {

        private List<Integer> queue;

        public MyStack() {
            queue = new ArrayList<>();
        }

        public void push(int x) {
            queue.add(x);
        }

        public int pop() {
            int val = queue.get(queue.size() - 1);
            queue.remove(queue.size() - 1);
            return val;
        }

        public int top() {
            return queue.get(queue.size() - 1);
        }

        public boolean empty() {
            return queue.isEmpty();
        }
    }

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */